mirrord/layer/src/lib.rs

```diff
@@ -495,8 +498,13 @@ pub(crate) unsafe extern "C" fn fork_detour() -> pid_t {
     match res.cmp(&0) {
         Ordering::Equal => {
             tracing::debug!("Child process initializing layer.");
-            let parent_connection = unsafe { PROXY_CONNECTION.take() }
-                .expect("parent connection doesn't exist in fork");
+            let parent_connection = match unsafe { PROXY_CONNECTION.take() } {
+                Some(conn) => conn,
+                None => {
+                    tracing::debug!("Skipping new inptroxy connection (trace only)");
+                    return res;
+                }
+            };
```

Comment:
Nitpick: I think this could be a let-else:

```suggestion
            let Some(parent_connection) = unsafe { PROXY_CONNECTION.take() } else {
                tracing::debug!("Skipping new inptroxy connection (trace only)");
                return res;
            };
```

Reply:
Yeah, this gives me a syntax error because of the `unsafe` block :x